examples/cloud-composer-datafusion-ingestion-templates/composer/dag-dependencies/read_params.py

```diff
@@ -0,0 +1,286 @@
+                #         env_param["audit_dataset"]+"_"+self.env_name
+                # else:
+                #     self.audit_dataset = env_param["audit_dataset"]
+                
+                self.audit_dataset = env_param["audit_dataset"]
+
+                self.param_file_path = env_param["param_file_path"]
+
+                dag_param_file_path = self.param_file_path \
+                    + system+"/"+ dag_param_file_nm
+                task_param_file_path= self.param_file_path \
+                    + system+"/"+ task_param_file_nm
+
+                logging.debug(f"dag_param_file_path: {dag_param_file_path}")
+                logging.debug(f"task_param_file_path: {task_param_file_path}")
+
+                self.data_bucket = env_param["data_bucket"]
+                self.archival_bucket = env_param["archival_bucket"]
+                self.composer_bucket = env_param["composer_bucket"]
+                self.cdf_instance_name = env_param["cdf_instance_name"]
+                self.cdf_namespace = env_param["cdf_namespace"]
+        except IOError:
+            logging.error(f"Error opening env_param_file {self.env_param_file}")
+            raise
+
+        # Read dag parameter file and set parameter values
+
+        logging.debug(f"Reading dag parameter file {dag_param_file_nm}")
+        try:
+            with open(dag_param_file_path) as dag_param_json:
+                dag_param = json.load(dag_param_json)
+
+                self.metadata_path = system
+                self.cdf_pipeline_timeout = dag_param["cdf_pipeline_timeout"]
+                self.update_metadata = dag_param["update_metadata"]
+                self.refresh_authorized_views = dag_param["refresh_authorized_views"] # pylint: disable=line-too-long
+                self.source_type = dag_param["source_type"]
+
+                if "load_type" not in dag_param:
+                    if self.source_type == "gcs":
+                        self.dag_load_type = "new_table"
+                    # elif self.source_type == "relational":
+                    #     self.dag_load_type = "SCD2"
+                    else:
+                        logging.error(
+                            f"load_type is not found in dag_param, "
+                            f"default load_type for {self.source_type} "
+                            f"not known. Specify load_type parameter "
+                            f"in DAG parameter file.")
+                else:
+                    self.dag_load_type = dag_param["load_type"]
+
+                # if self.env_name == "dev" or self.env_name == "test":
+                #     self.target_dataset = dag_param["target_dataset"]\
+                #         +"_"+self.env_name
+                # else:
+                #     self.target_dataset = dag_param["target_dataset"]
+                
+                self.target_dataset = dag_param["target_dataset"]
+
+                if not "raw" in self.target_dataset:
```

Comment:
this is not a good check for the error message you are rasing.
For example you will not raise on the dataset id `gcp_draw` but you should be cause it doesn't end in `_raw`
```suggestion
                if not self.target_dataset.endwith('_raw'):
```